src/python/zquantum/core/measurement.py

```diff
@@ -392,3 +393,96 @@ def convert_bitstring_to_int(bitstring: Iterable[int]) -> int:
             significant (little endian).
     """
     return int("".join(str(bit) for bit in bitstring[::-1]), 2)
+
+class Measurements:
+    """ A class representing measurements from a quantum circuit. The bitstrings variable represents the internal
+    data structure of the Measurements class. It is expressed as a list of tuples wherein each tuple is a measurement
+    and the value of the tuple at a given index is the measured bit-value of the qubit (indexed from N-1 -> 0) """
+
+    def __init__(self, bitstrings: Optional[List[Tuple[int]]] = None):
+        if bitstrings is None:
+            self.bitstrings = []
+        else:
+            self.bitstrings = bitstrings
+
+    @classmethod
+    def from_counts(cls, counts: Dict):
+        """ Create an instance of the Measurements class from a dictionary
+        
+        Args:
+            counts (dict): mapping of bitstrings to integers representing the number of times the bitstring was measured
+        """
+        measurements = cls()
+        measurements.add_counts(counts)
+        return measurements
+
+    @classmethod
+    def load_from_file(cls, file: TextIO):
+        """ Load a set of measurements from file 
+        
+        Args:
+            file (str or file-like object): the name of the file, or a file-like object
+        """        
+        if isinstance(file, str):
+            with open(file, 'r') as f:
+                data = json.load(f)
+        else:
+            data = json.load(file)
+
+        bitstrings = []
+        for bitstring in data["bitstrings"]:
+            bitstrings.append(tuple(bitstring))
+
+        return cls(bitstrings=bitstrings)
+
+    def save(self, filename: String):
+        """ Serialize the Measurements object into a file in JSON format.
+        
+        Args:
+            filename (string): filename to save the data to 
+        """
+        data = { "schema":     SCHEMA_VERSION+"-measurements",
+                 "counts":     self.get_counts(),
+                 "bitstrings": self.bitstrings}
+        with open(filename, "w") as f:
+            f.write(json.dumps(data, indent=2))
+
+
+    def get_counts(self):
+        """ Get the measurements as a histogram
+
+        Returns:
+            A dictionary mapping bitstrings to integers representing the number of times the bitstring was measured
+        """
+        bitstrings = convert_tuples_to_bitstrings(self.bitstrings)
+        return dict(Counter(bitstrings))
+
+
+    def add_counts(self, counts: Dict):
```

Comment:
FYI you an specify what should be the form of the Dict – e.g. Dict[Tuple[int], int] .
Not necessary to do so though.

Reply:
Good to know!